### ISSUE TITLE
Unable to deploy on vercel

### ISSUE THREAD

**Issue by ravencode** [Timilehin-bello/cycloneNFT#1]
[21:32:32.916] Running build in Cleveland, USA (East) – cle1
[21:32:32.968] Cloning github.com/ravencode/Metail-NFT (Branch: master, Commit: bb84cf3)
[21:32:33.214] Previous build cache not available
[21:32:33.602] Cloning completed: 633.097ms
[21:32:33.695] Running "vercel build"
[21:32:34.133] Vercel CLI 28.16.12
[21:32:34.377] Installing dependencies...
[21:32:34.669] yarn install v1.22.17
[21:32:34.692] info No lockfile found.
[21:32:34.698] [1/4] Resolving packages...
[21:32:36.234] warning ether > superagent@0.16.0: Please upgrade to v7.0.2+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>.
[21:32:36.267] warning ether > superagent > formidable@1.0.14: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
[21:32:46.363] [2/4] Fetching packages...
[21:32:59.088] [3/4] Linking dependencies...
[21:32:59.091] warning "web3modal > styled-components@5.3.8" has unmet peer dependency "react-is@>= 16.8.0".
[21:32:59.092] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@ethersproject/abi@^5.4.7".
[21:32:59.092] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@ethersproject/providers@^5.4.7".
[21:32:59.092] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@nomicfoundation/hardhat-network-helpers@^1.0.0".
[21:32:59.092] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@nomicfoundation/hardhat-chai-matchers@^1.0.0".
[21:32:59.093] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@nomiclabs/hardhat-ethers@^2.0.0".
[21:32:59.093] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@nomiclabs/hardhat-etherscan@^3.0.0".
[21:32:59.093] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@types/chai@^4.2.0".
[21:32:59.093] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@types/mocha@>=9.1.0".
[21:32:59.093] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@types/node@>=12.0.0".
[21:32:59.093] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "@typechain/hardhat@^6.1.2".
[21:32:59.093] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "chai@^4.2.0".
[21:32:59.093] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "ethers@^5.4.7".
[21:32:59.094] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "hardhat-gas-reporter@^1.0.8".
[21:32:59.094] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "solidity-coverage@^0.8.1".
[21:32:59.094] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "ts-node@>=8.0.0".
[21:32:59.094] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "typechain@^8.1.0".
[21:32:59.097] warning " > @nomicfoundation/hardhat-toolbox@2.0.2" has unmet peer dependency "typescript@>=4.5.0".
[21:33:08.928] [4/4] Building fresh packages...
[21:33:09.257] success Saved lockfile.
[21:33:09.262] Done in 34.60s.
[21:33:09.295] Detected Next.js version: 12.2.5
[21:33:09.296] Running "yarn run build"
[21:33:09.467] yarn run v1.22.17
[21:33:09.494] $ next build
[21:33:09.925] info  - SWC minify release candidate enabled. https://nextjs.link/swcmin
[21:33:09.960] Attention: Next.js now collects completely anonymous telemetry regarding usage.
[21:33:09.961] This information is used to shape Next.js' roadmap and prioritize features.
[21:33:09.961] You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
[21:33:09.961] https://nextjs.org/telemetry
[21:33:09.961] 
[21:33:10.091] info  - Linting and checking validity of types...
[21:33:10.304] warn  - No ESLint configuration detected. Run next lint to begin setup
[21:33:10.310] info  - Creating an optimized production build...
[21:33:16.085] Failed to compile.
[21:33:16.086] 
[21:33:16.086] ./NFTDetailsPage/NFTDescription/NFTDescription.jsx
[21:33:16.086] Module not found: Can't resolve '../../context/NFTMarketplaceContext' in '/vercel/path0/NFTDetailsPage/NFTDescription'
[21:33:16.086] 
[21:33:16.086] Import trace for requested module:
[21:33:16.086] ./NFTDetailsPage/NFTDetailsIndex.js
[21:33:16.086] ./NFTDetailsPage/NFTDetailsPage.jsx
[21:33:16.086] ./pages/NFT-details.js
[21:33:16.086] 
[21:33:16.086] ./components/Error/Error.jsx
[21:33:16.086] Module not found: Can't resolve '../../context/NFTMarketplaceContext' in '/vercel/path0/components/Error'
[21:33:16.086] 
[21:33:16.086] Import trace for requested module:
[21:33:16.087] ./components/componentsindex.js
[21:33:16.087] ./pages/_app.js
[21:33:16.087] 
[21:33:16.087] ./components/NFTCard/NFTCard.jsx
[21:33:16.087] Module not found: Can't resolve '../../context/NFTMarketplaceContext' in '/vercel/path0/components/NFTCard'
[21:33:16.087] 
[21:33:16.087] Import trace for requested module:
[21:33:16.087] ./components/componentsindex.js
[21:33:16.087] ./pages/_app.js
[21:33:16.087] 
[21:33:16.087] ./components/NavBar/NavBar.jsx
[21:33:16.087] Module not found: Can't resolve '../../context/NFTMarketplaceContext' in '/vercel/path0/components/NavBar'
[21:33:16.087] 
[21:33:16.087] ./components/componentsindex.js
[21:33:16.087] ./pages/_app.js
[21:33:16.087] 
[21:33:16.087] ./components/Success/Success.jsx
[21:33:16.087] Module not found: Can't resolve '../../context/NFTMarketplaceContext' in '/vercel/path0/components/Success'
[21:33:16.087] 
[21:33:16.087] Import trace for requested module:
[21:33:16.089] ./components/componentsindex.js
[21:33:16.089] ./pages/_app.js
[21:33:16.089] 
[21:33:16.089] 
[21:33:16.089] > Build failed because of webpack errors
[21:33:16.116] error Command failed with exit code 1.
[21:33:16.116] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[21:33:16.134] Error: Command "yarn run build" exited with 1

**Comment by ravencode** [comment]
Can you give me the steps to deploy this on vercel without any errors.

**Comment by Timilehin-bello** [comment]
There's a problem with the webpack  installation. 

So try to delete the package-log.json and node modules. 



When you're done deleting it, run npm install after it's done installing  try running vercel --prod or just vercel command


It should work

**Comment by ravencode** [comment]
Fixed it, there were 2 main problems in this line - import { NFTMarketplaceContext } from "../context/NFTMarketplaceContext";

1. lower case "c" used instead of "C"
2. "NFTMarketplaceContext" used instead of "NFTMarketplaceContext.js"

So. I changed this line - import { NFTMarketplaceContext } from "../context/NFTMarketplaceContext"; 
to 
import { NFTMarketplaceContext } from "../Context/NFTMarketplaceContext.js";

After these changes, all the errors were resolved and I could deploy it to vercel.
